### PR TITLE
regarding #276: try to fix Error: [] operator not supported for strings

### PR DESCRIPTION
**Pull request by k00ni**
In #276 the user @siwinski reported, that EasyRdf fails in php 7.1. I adapted the Serialiser/GraphViz.php::**escapeAttributes**:
- in this function $items was used as an array, but was initialized as string (fixed by setting $items = array())
- using $items[] = '...' seems failing in PHP 7.1, therefore using array_push to add string-items

In my local tests (running make test) with PHP 7.1.5 i got rid of the error `Error: [] operator not supported for strings`. Unfortunately, other tests seemed to fail now, but i guess, its for a different reason. For instance:

```
2) EasyRdf\Examples\VillagesTest::testCeres
Exception: Failed to run script (255): PHP Fatal error:  Uncaught EasyRdf\Http\Exception: HTTP request for http://www.dbpedialite.org/things/934787 failed: Bad Request in /var/www/html/vendor/easyrdf/easyrdf/lib/Graph.php:337
Stack trace:
#0 /var/www/html/vendor/easyrdf/easyrdf/lib/Graph.php(115): EasyRdf\Graph->load('http://www.dbpe...', NULL)
#1 /var/www/html/vendor/easyrdf/easyrdf/examples/villages.php(31): EasyRdf\Graph::newAndLoad('http://www.dbpe...')
#2 /var/www/html/vendor/easyrdf/easyrdf/test/cli_example_wrapper.php(43): require('/var/www/html/v...')
#3 {main}
  thrown in /var/www/html/vendor/easyrdf/easyrdf/lib/Graph.php on line 337
<html>
<head><title>EasyRdf Village Info Example</title></head>
<body>
<h1>EasyRdf Village Info Example</h1>
```

Nevertheless, this change may be welcome though.

### CODE REVIEW

**Comment by njh** [comment]
Thanks for this Pull Request.

Is there any chance you could add a test?

**Comment by k00ni** [comment]
Good morning. How would you test this protected function? My suggestion would be to create a dummy class with a proxy function, which calls `escapeAttributes`, just for test purposes.

File: lib/Serialiser/GraphViz.php [hunk 1]
```diff
@@ -223,9 +223,9 @@ protected function escape($input)
      */
     protected function escapeAttributes($array)
     {
-        $items = '';
+        $items = array();
         foreach ($array as $k => $v) {
-            $items[] = $this->escape($k).'='.$this->escape($v);
+            array_push($items, $this->escape($k).'='.$this->escape($v));
```
Comment (alexpott) [inline lib/Serialiser/GraphViz.php]: Why the change to array_push()? As the PHP docs say 

> Note: If you use array_push() to add one element to the array it's better to use $array[] = because in that way there is no overhead of calling a function.

http://us2.php.net/array_push

**Comment by alexpott** [comment]
@njh if the change was just $items = array() would you still want a test? Or is it just as simple to add PHP7.1 to https://github.com/njh/easyrdf/blob/master/.travis.yml?

**Comment by njh** [comment]
My goal for a test that verifies a bug fix is that:
* Write a test the fails and demonstrates that there is a bug
* Apply a fix for the bug
* Verify that the test now passes

So yes, in the case of something that is only a problem with a recent version of PHP, just adding PHP 7.1 to Travis should be sufficient. I will do that now...

**Comment by alexpott** [comment]
So it looks as though \EasyRdf\Serialiser\GraphVizTest is not running on travis because dot is not installed.

**Comment by njh** [comment]
I have cherry-picked 4ed264051ed407d59c2fde4128be176e96e8f22a by @alexpott that fixes this.

Tests are now passing for 7.1, 7.0, 5.6, 5.5, 5.4.
